### DISCUSSION TITLE
Multiple SRC and DST addresses

### DISCUSSION THREAD

**Discussion by justbruce** [esphome-econet/esphome-econet#244] [Ideas]
Currently, only a the 0x340 and 0x380 addresses are listened for or written to for HVAC.  More data is available if the 0x400 and 0x1c0 addresses are included.

  Is this even possible with yaml?  I'm not familiar enough to know.

**Comment by Daniel-dev22** [comment]
> Currently, only a the 0x340 and 0x380 addresses are listened for or written to for HVAC.  More data is available if the 0x400 and 0x1c0 addresses are included.
> 
>   Is this even possible with yaml?  I'm not familiar enough to know.

Sensors are being passively read from both units already there's no need to request directly for the majority of sensors as the thermostat is already doing that.

Also writing to those devices will never be formally supported as that's something only the thermostat should do.

**Comment by justbruce** [comment]
There are datapoints that are produced only when requested, and there are also datapoints that are  only requested when the thermostat is booting.

Why wouldn't we add the capability to initiate those requests?  We already imitate the thermostat when sending setpoint data.

**Reply by Daniel-dev22** [reply]
>  There are datapoints that are produced only when requested, and there are also datapoints that are  only requested when the thermostat is booting.
> 
> Why wouldn't we add the capability to initiate those requests?  We already imitate the thermostat when sending setpoint data.

I think you're misunderstanding how the component works. We listen to all requests on the bus.
 
We write to 0x380 as coming from 0x340 that's all. 

Any messages exchanged on the bus are listened to and can be brought into the component regardless of their source or destination address. 

For example, the ODU or the different furnace sensors for example are coming from devices replying to the thermostats request and we are reading that into the component already.

In the future it might be possible to request specific data points or an entire group of requests from a specific device other than the thermostat as it may give us data the thermostat never requests but I don't think we will ever support writing to devices on the bus other than the thermostat.
